backend/compound_soaking/src/graphql/soak_compound_res.rs

```diff
@@ -0,0 +1,159 @@
+            .all(db)
+            .await?)
+    }
+}
+
+#[ComplexObject]
+impl CompoundInstances {
+    /// Fetches all the crystals soaked with the compounds
+    async fn crystal_soaked(
+        &self,
+        ctx: &Context<'_>,
+    ) -> async_graphql::Result<Vec<soak_compound::Model>> {
+        let db = ctx.data::<DatabaseConnection>()?;
+        Ok(soak_compound::Entity::find()
+            .filter(soak_compound::Column::CompoundPlateId.eq(self.plate_id))
+            .filter(soak_compound::Column::CompoundWellNumber.eq(self.well_number))
+            .all(db)
+            .await?)
+    }
+}
+
+#[ComplexObject]
+impl soak_compound::Model {
+    /// Fetches the information of the crystals soaked from
+    /// crystal library subgraph
+    async fn crystals(&self) -> async_graphql::Result<CrystalWells> {
+        Ok(CrystalWells {
+            plate_id: self.crystal_plate_id,
+            well_number: self.crystal_well_number,
+        })
+    }
+
+    /// Fetches the information of the compounds soaked from
+    /// compound library subgraph
+    async fn compounds(&self) -> async_graphql::Result<CompoundInstances> {
+        Ok(CompoundInstances {
+            plate_id: self.compound_plate_id,
+            well_number: self.compound_well_number,
+        })
+    }
+}
+
+#[Object]
+impl SoakCompoundQuery {
+    /// Fetches all soaked compounds from the database
+    async fn soaked_compounds(
+        &self,
+        ctx: &Context<'_>,
+        cursor: CursorInput,
+    ) -> async_graphql::Result<ModelConnection<soak_compound::Model>> {
+        subject_authorization!("xchemlab.compound_soaking.read_soaked_compound", ctx).await?;
+        let db = ctx.data::<DatabaseConnection>()?;
+        Ok(cursor
+            .try_into_query_cursor::<soak_compound::Entity>()?
+            .all(db)
+            .await?
+            .try_into_connection()?)
+    }
+
+    /// Fetches a single soaked compound from the database
+    async fn soaked_compound(
+        &self,
+        ctx: &Context<'_>,
+        compound_plate_id: Uuid,
+        compound_well_number: i16,
+        crystal_plate_id: Uuid,
+        crystal_well_number: i16,
+    ) -> async_graphql::Result<Option<soak_compound::Model>> {
+        subject_authorization!("xchemlab.compound_soaking.read_soaked_compound", ctx).await?;
+        let db = ctx.data::<DatabaseConnection>()?;
+        Ok(soak_compound::Entity::find()
+            .filter(soak_compound::Column::CompoundPlateId.eq(compound_plate_id))
+            .filter(soak_compound::Column::CompoundWellNumber.eq(compound_well_number))
+            .filter(soak_compound::Column::CrystalPlateId.eq(crystal_plate_id))
+            .filter(soak_compound::Column::CrystalWellNumber.eq(crystal_well_number))
+            .one(db)
+            .await?)
+    }
+
+    /// Reference resolver for crystal wells
+    #[graphql(entity)]
+    async fn get_crystal_well_by_plate_id(&self, plate_id: Uuid, well_number: i16) -> CrystalWells {
```

Comment:
Would `related_x_y_z` be a more descriptive name for these resolvers?

Reply:
They are for the router just to identify an instance of a type using the keys (plate_id and well_number), they don't directly define any relation.

Reply:
Gotcha. Perhaps `route_crystal_well_by_location` or something?